packages/react-instantsearch/src/connectors/connectStateResults.js

```diff
@@ -0,0 +1,62 @@
+import createConnector from '../core/createConnector';
+import { getResults } from '../core/indexUtils';
+
+/**
+ * The `connectStateResults` connector provides a way to access the `searchState` and the `searchResults`
+ * of InstantSearch.
+ * For instance this connector allows you to create results/noResults or query/noQuery pages.  
+ * @name connectStateResults
+ * @kind connector
+ * @providedPropType {object} searchState - The search state of the instant search component. <br/><br/> See: [Search state structure](https://community.algolia.com/react-instantsearch/guide/Search_state.html)
+ * @providedPropType {object} searchResults - The search results. <br/><br/> In case of multiple indices: if used under `<Index>`, results will be those of the corresponding index otherwise it'll be those of the root index  See: [Search results structure](https://community.algolia.com/algoliasearch-helper-js/reference.html#searchresults)
+ * @providedPropType {object} allSearchResults - In case of multiple indices you can retrieve all the results 
+ * @providedPropType {string} error - If the search failed, the error will be logged here. 
+ * @providedPropType {boolean} searching - If there is a search in progress. 
+ * @providedPropType {boolean} searchingForFacetValues - If there is a search in a list in progress. 
+ * @example
+ * import React from 'react';
+ *
+ * import { InstantSearch, Hits } from 'react-instantsearch/dom';
+ * import { connectStateResults } from 'react-instantsearch/connectors';
+ *
+ * const Content = connectStateResults(
```

Comment:
I'm having a little doubts about the name here. I think we can find something that describes intent rather than what it is. 

`connectConditional` describes the use case better IMO

Reply:
This could be use in the future for future use case such as retrieving the `userInfo` field you can get with query rules (and probably a usage with analytics also). Then we need a name that could represent this. Cf discussion in the issue thread. If you have a better name you can still propose it.

Reply:
There's no API feature in this connector that has any conditional mean (no if/else) so naming it conditional would be not the best to me too.